src/workerd/io/tracer.c++

```diff
@@ -50,7 +50,8 @@ void TailStreamWriter::report(const InvocationSpanContext& context, TailEvent::E
     }
   }
 
-  tracing::TailEvent tailEvent(context, s.timeSource(), s.sequence++, kj::mv(event));
+  tracing::TailEvent tailEvent(context.getTraceId(), context.getInvocationId(), context.getSpanId(),
```

Comment:
Why?

Reply:
Here, we form a tail event using the new SpanContext as used in the JS API. This differs from the internal InvocationSpanContext, so we need to pass its fields individually.